src/screens/Products/AddProduct.js

```diff
@@ -0,0 +1,310 @@
+import { AddImage, AsyncButton, UserInput, Wallpaper } from '@components';
+import React, { Component } from 'react';
+import { StyleSheet, ScrollView, View, Text, TextInput, Picker } from 'react-native';
+import ImagePicker from 'react-native-image-crop-picker';
+
+export default class AddProduct extends Component {
+  static navigationOptions = {
+    title: 'Add a Handmade Product'
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      productMainCategory: 'Handmade Apparel',
+      productSubCategory: 'Activewear',
+
+      productMainPicture: "",
+      productTitle: "",
+      productDescription: "",
+      productGender: 'Women',
+      productStandardPrice: "",
+      productSellerSKU: "",
+      productQuantity: "",
+      productTime: "",
+
+      //If you adding product, assumes never sold before
+      productTimesSold: 0,
+      focusedInputs: {title: false, desc: false, price: false, sku: false, quantity: false, time: false},
+      adding: false
+    }
+
+    this.pickImage = this.pickImage.bind(this)
+    this.createProduct = this.createProduct.bind(this)
+    this.verifyFields = this.verifyFields.bind(this)
+  }
+
+  pickImage() {
+    ImagePicker.openPicker({
+      width: 300,
+      height: 300,
+      cropping: true
+    }).then(image => {
+      this.setState({productMainPicture: image.path})
+    });
+  }
+
+  verifyFields() {
```

Comment:
A bit clunky with all the conditionals and adds complexity, possibly look into a cleaner implementation involving a boolean and what field might be missing through some type of switch case etc.

Reply:
This serves the purpose of functionality so if an alternative isn't established then just keep the code. Similar conditions are found in the add artisan.